src/app/components/driver-register/driver-register.component.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, OnInit } from '@angular/core';
+import { UserService } from 'src/app/services/user-service/user.service';
+import { BatchService } from 'src/app/services/batch-service/batch.service';
+import { Batch } from 'src/app/models/batch';
+import { Router } from '@angular/router';
+
+@Component({
+	selector: 'app-driver-register',
+	templateUrl: './driver-register.component.html',
+	styleUrls: ['./driver-register.component.css']
+})
+export class DriverRegisterComponent implements OnInit {
+
+	batches: Batch[] = [];
+
+	userName: string = '';
+	firstName: string = '';
+	lastName: string = '';
+	email: string = '';
+	phone: string = '';
+	batchNum: number;
+
+	constructor(private userService: UserService, private batchService: BatchService, private router: Router) { }
+
+	ngOnInit() {
+		if (sessionStorage.getItem('auth')) {
+			this.router.navigate(['home']);
+		} else {
+			this.batchService.getAllBatches()
+				.subscribe(allBatches => {
+					this.batches = allBatches;
+					this.batchNum = this.batches[0].batchNumber;
+			});
+		}
+	}
+
+	validateUserName() {
+		return this.userName.length >= 3 && this.userName.length <= 8;
+	}
+
+	validateName(name: string) {
+		return /^([a-zA-z]){1,20}$/.test(name) && name.length < 20;
+	}
+
+	nameFormat(name: string) {
+		return name[0].toUpperCase() + name.slice(1).toLowerCase();
+	}
+
+	phoneFormat(phone: string) {
+		return phone.replace(/[^0-9]/g, '').replace(/(\d{3})(\d{3})(\d{4})/, "($1) $2-$3");
+	}
+
+	validateEmail() {
+		return /^\w+\.?\w+@\w+\.\w{2,4}$/.test(this.email);
+	}
+
+	validatePhone() {
```

Comment:
perhaps we could use a service here for validation rather than including this directly in the component